packages/dataviews/src/types.ts

```diff
@@ -207,6 +207,11 @@ export interface Filter {
 	 * The value to filter by.
 	 */
 	value: any;
+
+	/**
+	 * Whether the filter can be edited or removed.
+	 */
+	isLocked?: boolean;
```

Comment:
I considered making this part of the `fields.filterBy` config, as `isPrimary` is. I decided against because having it as a view filter property collocates both concerns together: declaring the values & declaring those values are locked.

This made me think that `fields.filterBy.isPrimary` is actually not collocated properly. It should be part of the View API, not of Field API: it is about what's visible or not in the UI, not about expressing what a field can do.

Reply:
This makes sense to me and the follow up you're suggesting.